### PR TITLE
Update version of ZuluFX January 2020 release

### PR DESCRIPTION
**Pull request by poad**

### CODE REVIEW

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 1]
```diff
@@ -2,6 +2,76 @@ package io.sdkman.changelogs.java
...
+        Some(Zulu)
+      )
+    ).validate().insert()
+  }
+
+  @ChangeSet(order = "0002", id = "0002-add_zulufx_11_0_6", author = "dimitryc")
+  def migrate0002(implicit db: MongoDatabase) = {
+    Seq(Linux64, Windows, MacOSX).foreach(
+      platform =>
+        removeVersion(candidate = "java", version = "11.0.2-zulu", platform)
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: Same as above.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 2]
```diff
@@ -2,6 +2,76 @@ package io.sdkman.changelogs.java
...
+    )
+    List(
+      Version(
+        "java",
+        "11_0_6-zulufx",
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: Not a valid version format.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 3]
```diff
@@ -2,6 +2,76 @@ package io.sdkman.changelogs.java
...
+      ),
+      Version(
+        "java",
+        "11_0_6-zulufx",
+        "https://cdn.azul.com/zulu/bin/zulu11.35.15-ca-fx-jdk11.0.5-macosx_x64.tar.gz",
+        MacOSX,
+        Some(Zulu)
+      )
+    ).validate().insert()
+    setCandidateDefault("java", "11_0_6-zulufx")
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: Zulu FX will never be the default Java.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 4]
```diff
@@ -2,6 +2,76 @@ package io.sdkman.changelogs.java
...
+import io.sdkman.changelogs.{AdoptOpenJDK, Linux64, MacOSX, Version, Windows, _}
 
 @ChangeLog(order = "018")
-class AzulZuluFxMigrations {}
+class AzulZuluFxMigrations {
+
+  @ChangeSet(
+    order = "0001",
+    id = "0001-add_zulufx_8_0_242",
+    author = "dimitryc"
```
[the remark this answers is not in the document]
Comment (poad) [reply src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: I'm sorry. I copied it, and a correction has leaked out from the others.
I make modifications.

**Comment by poad** [comment]
@marc0der 
I am sorry that I take out pull request very of bad quality.
I review it generally once again.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala (hunk 3 above)
Comment (poad) [reply src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: OK!
I removed the this line.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 5]
```diff
@@ -2,6 +2,76 @@ package io.sdkman.changelogs.java
...
+  )
+  def migrate0001(implicit db: MongoDatabase) = {
+    Seq(Linux64, Windows, MacOSX).foreach(
+      platform =>
+        removeVersion(candidate = "java", version = "8.0.202-zulu", platform)
```
[the remark this answers is not in the document]
Comment (poad) [reply src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: I made remove an independent change set.
And I made modifications in zulufx.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 6]
```diff
@@ -2,6 +2,76 @@ package io.sdkman.changelogs.java
...
+        "java",
+        "8.0.232-zulufx",
+        "https://cdn.azul.com/zulu/bin/zulu8.42.0.23-ca-fx-jdk8.0.232-macosx_x64.tar.gz",
+        MacOSX,
+        Some(Zulu)
+      )
+    ).validate().insert()
+  }
+
+  @ChangeSet(order = "0002", id = "0002-add_zulufx_11_0_6", author = "dimitryc")
```
[the remark this answers is not in the document]
Comment (poad) [reply src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: I made modifications.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala (hunk 1 above)
Comment (poad) [reply src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: I made remove an independent change set.
And I made modifications in zulufx.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala (hunk 2 above)
Comment (poad) [reply src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: I changed it to 11.0.5-zuluxf.

**Comment by poad** [comment]
@marc0der 
I reviewed it generally and made modifications.
Please review again.

**Comment by marc0der** [comment]
Thanks for making the changes @poad.

Something that I've been considering for some time is to change the version string for the ZuluFX distribution to be something like `11.0.5.fx-zulu` instead of `11.0.5-zulufx`. It will be more inline with what we have for AdoptOpenJDK HS and J9 distributions. wdyt?

**Comment by poad** [comment]
Thanks for reviewing my requests @marc0der.

It was zulufx until now.
It likes https://github.com/sdkman/sdkman-db-migrations/blob/master/src/main/scala/io/sdkman/changelogs/JavaMigrations.scala#L96-L148 .

I replace it with fx-zulu by this update. It like AdoptOpenJDK HS and J9 distributions and Liberica JDK with FX.
https://github.com/sdkman/sdkman-db-migrations/blob/master/src/main/scala/io/sdkman/changelogs/java/BellSoftLibericaMigrations.scala#L174-L206

We OK it, and let's change it.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 7]
```diff
@@ -2,6 +2,89 @@ package io.sdkman.changelogs.java
...
+      Version(
+        "java",
+        "8.0.232.fx-zulu",
+        "https://cdn.azul.com/zulu/bin/zulu8.42.0.23-ca-fx-jdk8.0.232-win_x64.zip",
+        Windows,
+        Some(Zulu)
+      ),
+      Version(
+        "java",
+        "8.0.232.fx--zulu",
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: `--`

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 8]
```diff
@@ -2,6 +2,89 @@ package io.sdkman.changelogs.java
...
+      )
+    ).validate().insert()
+  }
+
+  @ChangeSet(order = "0002", id = "0002-add_zulufx_11_0_5", author = "poad")
+  def migrate0002(implicit db: MongoDatabase) = {
+    List(
+      Version(
+        "java",
+        "11.0.5.fx--zulu",
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: Same.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 9]
```diff
@@ -2,6 +2,89 @@ package io.sdkman.changelogs.java
...
+        "https://cdn.azul.com/zulu/bin/zulu11.35.15-ca-fx-jdk11.0.5-linux_x64.tar.gz",
+        Linux64,
+        Some(Zulu)
+      ),
+      Version(
+        "java",
+        "11.0.5.fx--zulu",
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: Same.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 10]
```diff
@@ -2,6 +2,89 @@ package io.sdkman.changelogs.java
...
+        "https://cdn.azul.com/zulu/bin/zulu11.35.15-ca-fx-jdk11.0.5-win_x64.zip",
+        Windows,
+        Some(Zulu)
+      ),
+      Version(
+        "java",
+        "11.0.5.fx--zulu",
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: Same.

File: src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala [hunk 11]
```diff
@@ -2,6 +2,89 @@ package io.sdkman.changelogs.java
 
 import com.github.mongobee.changeset.{ChangeLog, ChangeSet}
 import com.mongodb.client.MongoDatabase
+import io.sdkman.changelogs.{ZuluFX, Linux64, MacOSX, Version, Windows, _}
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: I'm guessing that all these imports aren't really needed.

Comment (poad) [reply src/main/scala/io/sdkman/changelogs/java/AzulZuluFxMigrations.scala]: I resigned as a wild card and changed it to describe only a necessary article.

**Comment by poad** [comment]
@marc0der 
I'm sorry many times.
As I think that this time is all right, I would like a review once again.

**Comment by poad** [comment]
As a change was taken in, close makes it.

f2d70cdaf384918f59e53c9b66d70eb3b7305b0f
3a311fe86e3ed5c3aaa9f3bc82094ccb91be691b
f38746e29303027bfa1332b938bac861f6e4ce42
6029144256c323bb39da2b671e4c0e3350e65f2e
3b0258ba739f202a045ba50c1e4dd636e574a711
43a3f36110e222d31669b3bdbe924c328ea69e76
398aeeb59ab47bb6846fad8c195cfbb53a08c98f
b0308a9223ab144c49f25da22894eb7b207a411d

**Comment by marc0der** [comment]
I had to rebase against master but it is now merged.

**Pull request closed without merging**